modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java

```diff
@@ -636,46 +636,50 @@ protected IgniteCheckedException concurrentStateChangeError(boolean activate) {
 
     /** {@inheritDoc} */
     @Override public void collectGridNodeData(DiscoveryDataBag dataBag) {
-        if (!dataBag.commonDataCollectedFor(STATE_PROC.ordinal())) {
-            DiscoveryDataBag.JoiningNodeDiscoveryData joiningNodeData = dataBag.newJoinerDiscoveryData(STATE_PROC.ordinal());
+        DiscoveryDataBag.JoiningNodeDiscoveryData joiningNodeData = dataBag.newJoinerDiscoveryData(STATE_PROC.ordinal());
 
-            BaselineTopologyHistory historyToSend = null;
+        if (joiningNodeData != null && !joiningNodeData.hasJoiningNodeData())
+            compatibilityMode = true; //compatibility mode: only old nodes don't send any data on join
 
-            if (joiningNodeData != null) {
-                if (!joiningNodeData.hasJoiningNodeData() || compatibilityMode) {
-                    //compatibility mode: old nodes don't send any data on join, so coordinator of new version
-                    //doesn't send BaselineTopology history, only its current globalState
-                    dataBag.addGridCommonData(STATE_PROC.ordinal(), globalState);
+        if (dataBag.commonDataCollectedFor(STATE_PROC.ordinal()) || joiningNodeData == null)
+            return;
 
-                    return;
-                }
+        if (!joiningNodeData.hasJoiningNodeData() || compatibilityMode) {
+            //compatibility mode: old nodes don't send any data on join, so coordinator of new version
+            //doesn't send BaselineTopology history, only its current globalState
+            dataBag.addGridCommonData(STATE_PROC.ordinal(), globalState);
 
-                DiscoveryDataClusterState joiningNodeState = null;
+            return;
+        }
 
-                try {
-                    if (joiningNodeData.joiningNodeData() != null)
-                        joiningNodeState = marsh.unmarshal(
-                            (byte[]) joiningNodeData.joiningNodeData(),
-                            U.resolveClassLoader(ctx.config()));
-                } catch (IgniteCheckedException e) {
-                    U.error(log, "Failed to unmarshal disco data from joining node: " + joiningNodeData.joiningNodeId());
-
-                    return;
-                }
+        DiscoveryDataClusterState joiningNodeState = null;
+
+        try {
+            if (joiningNodeData.joiningNodeData() != null)
+                joiningNodeState = marsh.unmarshal(
+                    (byte[])joiningNodeData.joiningNodeData(),
+                    U.resolveClassLoader(ctx.config())
+                );
+        }
+        catch (IgniteCheckedException e) {
```

Comment:
Stacktrace of exception is not logged, eception  is ignored. Is it reasonable to log? Is it reasonable to rename to 'ignored'?